VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,184 @@
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ENDORSEMENT,
+                new StubAuthorData(author, label, null),
+                endorsePrefix + " " + "by author (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ENDORSEMENT,
+                new StubAuthorData(null, null, creationDate),
+                endorsePrefix + " " + relativeTime);
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ENDORSEMENT,
+                new StubAuthorData(author, null, null),
+                endorsePrefix + " " + "by author");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ENDORSEMENT,
+                new StubAuthorData(null, label, null),
+                endorsePrefix + " " + "(label)");
+
+        // For answer
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, creationDate),
+                answerPrefix + " " + relativeTime + " by author (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, creationDate),
+                answerPrefix + " " + relativeTime + " (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, creationDate),
+                answerPrefix + " " + relativeTime + " by author");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, null),
+                answerPrefix + " " + "by author (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, creationDate),
+                answerPrefix + " " + relativeTime);
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, null),
+                answerPrefix + " " + "by author");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, null),
+                answerPrefix + " " + "(label)");
+
+        // Empty Case
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, null),
+                null);
+    }
+
+    private void applyAssertions(DiscussionTextUtils.AuthorAttributionLabel type, IAuthorData input,
+                                 String expectedOutput) {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView, type, input, now,
+                dummyClickListener);
+        if (expectedOutput == null) {
+            assertTrue(dummyTextView.getVisibility() == View.GONE);
+        } else {
+            String output = dummyTextView.getText().toString();
+            assertEquals(expectedOutput, output);
+        }
+    }
+
+    @Test
+    public void testAuthorIsTappable() {
```

Comment:
If we got with the AOSP standard, it would be something like `testOnClick_authorAttributionText`